05/ContactForm.js

```diff
@@ -1,10 +1,86 @@
-import React from 'react';
+import React, { useReducer } from 'react';
 
-import account from './account';
 
 const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
+
+    const init = { firstName: '', lastName: '', email: '', tel: '', date: '', time: '', comment: '' };
+
+    const reducer = (state, {name, value}) => {
+        return { ...state, [name]: value};
+    }
+
+    const [state, dispatch] = useReducer(reducer, init);
+    const { firstName, lastName, email, tel, date, time, comment } = state;
+
+
+    const validationPhoneNumber = () => {
+        const reg = /([0-9]{3})-([0-9]{3})-([0-9]{3})/;
+        return reg.test(tel)
+    }
+
+    const validationData = () => {
+        const reg = /([0-9]{4})-([0-9]{2})-([0-9]{2})/;
+        return reg.test(date)
+    }
+
+    const validationTime = () => {
+        const reg = /([0-9]{2}):([0-9]{2})/;
+        return reg.test(time)
+    }
+
+    const validationForm = (errors) => {
+        if (firstName.length === 0 || firstName.length <= 1) {
+            errors.push('Field firstname is required! \n');
+        }
+        if (lastName.length === 0 || lastName.length <= 1) {
+            errors.push('Field lastname is required! \n');
+        }
+        if (!email.includes('@')) {
+            errors.push('Email need @ sign! \n');
+        }
+        if (!validationPhoneNumber()) {
+            errors.push('Phone number is incorrect! (Write e.g 500-500-500) \n');
+        }
+        if (!validationData()) {
+            errors.push('Date is incorrect! \n');
+        }
+        if (!validationTime()) {
+            errors.push('Time is incorrect! \n');
+        }
+        if (comment.length === 0 || comment.length <= 1) {
+            errors.push('Field comment is required! \n');
+        }
+    }
+
+    const handleSubmit = e => {
+        e.preventDefault();
+
+        const errors = [];
+        validationForm(errors);
+
+        if (errors.length > 0) {
+            window.alert(errors)
```

Comment:
Może warto by było użyć `errors.join('sepearator')`, aby ładniej to wyglądało :P